editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -2941,16 +2944,235 @@ void Node3DEditorViewport::_notification(int p_what) {
+						delta.z = 0.0;
+					}
+
+					int active_axes = 0;
+					if (Math::abs(delta.x) > 0.0) {
+						active_axes++;
+					}
+					if (Math::abs(delta.y) > 0.0) {
+						active_axes++;
+					}
+					if (Math::abs(delta.z) > 0.0) {
+						active_axes++;
+					}
+
+					String x_text = Math::abs(delta.x) > 0.0 ? TS->format_number(vformat("X: %.3f m", Math::abs(delta.x))) : "";
+					String y_text = Math::abs(delta.y) > 0.0 ? TS->format_number(vformat("Y: %.3f m", Math::abs(delta.y))) : "";
+					String z_text = Math::abs(delta.z) > 0.0 ? TS->format_number(vformat("Z: %.3f m", Math::abs(delta.z))) : "";
+
+					Vector3 corner_point;
+
+					if (active_axes >= 2) {
+						if (Math::abs(delta.x) > 0.0 && Math::abs(delta.y) > 0.0 && Math::abs(delta.z) == 0.0) {
+							// XY plane
+							corner_point = Vector3(end_pos.x, start_pos.y, start_pos.z);
+
+							ruler_label_x->set_text(x_text);
+							ruler_label_x->set_visible(true);
+							Vector2 x_pos = camera->unproject_position((start_pos + corner_point) / 2) - (ruler_label_x->get_custom_minimum_size() / 2);
+							ruler_label_x->set_position(x_pos);
+
+							ruler_label_y->set_text(y_text);
+							ruler_label_y->set_visible(true);
+							Vector2 y_pos = camera->unproject_position((corner_point + end_pos) / 2) - (ruler_label_y->get_custom_minimum_size() / 2);
+							ruler_label_y->set_position(y_pos);
+
+							ruler_label_z->set_visible(false);
+						} else if (Math::abs(delta.x) > 0.0 && Math::abs(delta.z) > 0.0 && Math::abs(delta.y) == 0.0) {
+							// XZ plane
+							corner_point = Vector3(end_pos.x, start_pos.y, start_pos.z);
+
+							ruler_label_x->set_text(x_text);
+							ruler_label_x->set_visible(true);
+							Vector2 x_pos = camera->unproject_position((start_pos + corner_point) / 2) - (ruler_label_x->get_custom_minimum_size() / 2);
+							ruler_label_x->set_position(x_pos);
+
+							ruler_label_y->set_text(z_text);
+							ruler_label_y->set_visible(true);
+							Vector2 z_pos = camera->unproject_position((corner_point + end_pos) / 2) - (ruler_label_y->get_custom_minimum_size() / 2);
+							ruler_label_y->set_position(z_pos);
+
+							ruler_label_z->set_visible(false);
+						} else if (Math::abs(delta.y) > 0.0 && Math::abs(delta.z) > 0.0 && Math::abs(delta.x) == 0.0) {
+							// YZ plane
+							corner_point = Vector3(start_pos.x, start_pos.y, end_pos.z);
+
+							ruler_label_x->set_text(z_text);
+							ruler_label_x->set_visible(true);
+							Vector2 z_pos = camera->unproject_position((start_pos + corner_point) / 2) - (ruler_label_x->get_custom_minimum_size() / 2);
+							ruler_label_x->set_position(z_pos);
+
+							ruler_label_y->set_text(y_text);
+							ruler_label_y->set_visible(true);
+							Vector2 y_pos = camera->unproject_position((corner_point + end_pos) / 2) - (ruler_label_y->get_custom_minimum_size() / 2);
+							ruler_label_y->set_position(y_pos);
+
+							ruler_label_z->set_visible(false);
+						} else if (active_axes == 3) {
```

Comment:
```suggestion
						} else {
```
Considering the previous checks and the block we are in, I don't see how you can get to this check and have it come up false.

Reply:
Agreed, done.